---
title: "Sled_dog_manuscript"
output:
  pdf_document: default
  html_document: default
date: "2023-03-03"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Data preparation

## Load required libraries

```{r libraries}
library(vegan)
library(phyloseq)
library(hilldiv)
library(ape)
library(phytools)
library(dplyr)
library(tibble)
library(microbiome)
library(ggplot2)
library(RColorBrewer)
library(ggpubr)
library(ranacapa)
#library(phylox)
library(lme4)
library(MuMIn)
library(sjPlot)
library(nlme)
library(matrixStats)
library(microbiomeutilities)
library(iNEXT)
library(tidyverse)
library(dplyr)
library(janitor)
library(ggh4x)
library(tinytable)
library(hilldiv2)
```

## Declare directories and files

```{r directories}
physeq_phylum_clean <- readRDS("resources/physeq_phylum_filtered_tree.RData")
physeq_phylum_clean <- prune_taxa(taxa_sums(physeq_phylum_clean)>0, physeq_phylum_clean)
physeq_FL <- subset_samples(physeq_phylum_clean, Working_status %in% c("Follower", "Lead"))
physeq_FL <- prune_taxa(taxa_sums(physeq_FL)>0, physeq_FL)
physeq_FL.rel <-  microbiome::transform(physeq_FL, "compositional")
physeq_FL.rel = filter_taxa(physeq_FL.rel, function(x) sum(x) > 0.005, TRUE)
a <- data.frame(physeq_FL.rel@otu_table)
b <- data.frame(physeq_FL@otu_table)
c <- b %>% filter(row.names(b) %in% row.names(a))

d <- otu_table(c, taxa_are_rows=T)
otu_table(physeq_FL) <- d

metadata <- data.frame(physeq_FL@sam_data)
metadata$Pack=factor(metadata$Pack)
metadata$Group <- paste("Group", metadata$Pack, sep = "_")
metadata_row <- rownames_to_column(metadata, "sample")
sample_data(physeq_FL) <- metadata

asv_table <- data.frame(physeq_FL@otu_table) %>% rownames_to_column(., "asv")
taxonomy_table <- data.frame(physeq_FL@tax_table)%>% rownames_to_column(., "asv")
tree = phy_tree(physeq_FL)
tree <- force.ultrametric(tree, method = "extend")
asv_table_rel <- asv_table %>%
  mutate_at(vars(-asv),~./sum(.))


# physeq_filtered_keep <- readRDS("/Users/jtk712/Documents/Projects/GreenDogs/DADA2/Seq1/physeq_filtered_keepNO.RData")
# physeq_bacteria <- subset_taxa(physeq_filtered_keep, Kingdom == "Bacteria")
# physeq_others <- subset_taxa(physeq_filtered_keep, Kingdom != "Bacteria")#35
#physeq_phylum <- prune_taxa(taxa_sums(physeq_phylum)>0, physeq_phylum)
```


## Summary

### Explore the number of samples 
```{r samples_all, comment="", echo=FALSE}
table(meta(physeq_FL)$Extraction, meta(physeq_FL)$Location)
table(meta(physeq_FL)$Working_status, meta(physeq_FL)$Location)
table(meta(physeq_FL)$Extraction)
```

### Rarefraction curves

```{r rarecurves}
#pdf("figures/rarefaction.pdf",width=14, height=9)
ggrare(physeq_FL, step = 1000, color = "Extraction", se = FALSE) +
  scale_color_manual(values=c("#e5bd5b", "#6b7398"))+
  theme_classic() + 
  labs(colour = "Body site")
#dev.off()
```

```{r rare_species, comment="", echo=FALSE, message=FALSE, warning=FALSE}
E_samples = row.names(metadata[metadata$Extraction == "Saliva",])
H_samples = row.names(metadata[metadata$Extraction == "Faecal",])

#Subset countable
countable_E <- asv_table[,colnames(asv_table) %in% E_samples]
countable_E[countable_E > 0] <- 1
countable_H <- asv_table[,colnames(asv_table) %in% H_samples]
countable_H[countable_H > 0] <- 1

#Add them to a list
countlist <- list(Saliva=countable_E,Faecal=countable_H)
out <- iNEXT(countlist, q=0, datatype="incidence_raw")
ggiNEXT(out)
```

### All ASVs
```{r allasv, comment="", echo=FALSE, message=FALSE, warning=FALSE}
cat(nrow(physeq_FL@tax_table))
```


### Bacteria with class information
```{r bact_class, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_class <- subset_taxa(physeq_FL, !is.na(Class))
cat(nrow(physeq_class@tax_table))
```
```{r bact_class_info, comment="", echo=FALSE, message=FALSE, warning=FALSE}
table(tax_table(physeq_class)[,"Class"]) %>% as.data.frame() %>% arrange(desc(Freq))
```

## Bacteria without class information
```{r bact_noclass, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_classNA <- subset_taxa(physeq_FL, is.na(Class))#all have class information
```

## Number of unique Phyla
```{r bact_phyla, comment="", echo=FALSE, message=FALSE, warning=FALSE}
phyla <- get_taxa_unique(physeq_FL, "Phylum")
cat(length(phyla))
```

## Number of unique orders
```{r bact_order, comment="", echo=FALSE, message=FALSE, warning=FALSE}
order <- get_taxa_unique(physeq_FL, "Order")
cat(length(order))
```

## Number of unique families
```{r bact_fam, comment="", echo=FALSE, message=FALSE, warning=FALSE}
family <- get_taxa_unique(physeq_FL, "Family")
cat(length(family))
```

## Number of unique genera
```{r bact_gene, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genus <- get_taxa_unique(physeq_class, "Genus")
cat(length(genus))
```

## Faecal vs oral samples

```{r faecal, warning=FALSE, echo=FALSE}
physeq_faecal <- subset_samples(physeq_FL, Extraction == "Faecal")
physeq_faecal <- prune_taxa(taxa_sums(physeq_faecal)>0, physeq_faecal)
faecal_table <- data.frame(physeq_faecal@otu_table)
faecal_metadata <- data.frame(physeq_faecal@sam_data)
faecal_metadata$Pack=factor(faecal_metadata$Pack)
faecal_metadata$Group <- paste("Group", faecal_metadata$Pack, sep = "_")
sample_data(physeq_faecal) <- faecal_metadata
tree_faecal = phy_tree(physeq_faecal)
hierarchy_faecal <- tibble::rownames_to_column(faecal_metadata, "Sample")
hierarchy_faecal <- hierarchy_faecal[which(hierarchy_faecal[,1] %in% colnames(faecal_table)),]
```

```{r saliva}
physeq_saliva <- subset_samples(physeq_FL, Extraction == "Saliva")
physeq_saliva <- prune_taxa(taxa_sums(physeq_saliva)>0, physeq_saliva)
saliva_table <- data.frame(physeq_saliva@otu_table)
saliva_metatable <- data.frame(physeq_saliva@sam_data)
saliva_taxonomy <- data.frame(physeq_saliva@tax_table)%>% rownames_to_column(., "asv")
tree_saliva = phy_tree(physeq_saliva)
tree_saliva <- force.ultrametric(tree_saliva, method = "extend")
identical(sort(colnames(saliva_table)),sort(as.character(saliva_metatable[,1])))
match_data(saliva_table,tree_saliva)
saliva_table <- data.frame(physeq_saliva@otu_table)%>% rownames_to_column(., "asv")
saliva_metatable$Pack=factor(saliva_metatable$Pack)
saliva_metatable$Group <- paste("Group", saliva_metatable$Pack, sep = "_")
sample_data(physeq_saliva) <- saliva_metatable
saliva_metatable<- saliva_metatable %>% rownames_to_column(., "sample")
```

## Number of unique Phyla in saliva
```{r bact_phyla, comment="", echo=FALSE, message=FALSE, warning=FALSE}
get_taxa_unique(physeq_saliva, "Phylum")
```

## Number of unique Phyla in faeces
```{r bact_phyla, comment="", echo=FALSE, message=FALSE, warning=FALSE}
get_taxa_unique(physeq_faecal, "Phylum")
```

## Barplots
```{r barplot_na, comment="", echo=FALSE, message=FALSE, warning=FALSE}
top_phylum <- aggregate_top_taxa2(physeq_FL, top = 10, "Phylum")
metadata_row_na <- data.frame(sample_data(top_phylum)) %>% rownames_to_column(., "sample")
asv_table_na <- data.frame(top_phylum@otu_table)%>% rownames_to_column(., "asv")
taxonomy_table_na <- data.frame(top_phylum@tax_table)%>% rownames_to_column(., "asv")


#pdf("figures/Barplot_all_filtered.pdf",width=14, height=9)
asv_table_na %>%
  mutate_at(vars(-asv),~./sum(.)) %>% 
  pivot_longer(-asv, names_to = "sample", values_to = "count") %>% 
  left_join(., taxonomy_table_na, by = join_by(asv == asv)) %>% 
  left_join(., metadata_row_na, by = join_by(sample == sample)) %>% 
  ggplot(., aes(x=sample,y=count, fill=fct_reorder(Phylum,count))) + 
  geom_bar(stat="identity") +#, colour="white", linewidth=0.1
  scale_fill_manual(values = c("#000000", "#3c634e","#C0C0C0", "#98AFC7", "#6698FF", "#153E7E","#e5bd5b", "#6b7398", "#76b183", "#d57d2c", "#2a2d26"))+
  facet_grid(.~Extraction,  scales="free_x")+
theme(strip.background = element_blank(),
      axis.text.x=element_blank(),
      axis.title.x = element_blank(),
      axis.ticks.x=element_blank(),
      axis.title = element_text(size = 10, face = "bold"),
      panel.background = element_blank(),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 10),
      legend.title = element_text(size=12, face = "bold"),
      legend.key.size = unit(0.3, 'cm'),
      strip.text.x = element_text(size = 12, colour = "black")) +
  labs(x = "Sample", y = "Relative abundances",fill="Top 10 phyla")+
  guides(fill = guide_legend(ncol = 1))
#dev.off()
```

### Which phyla are different in both sample types

```{r phyl_MW, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_clr <- microbiome::transform(physeq_FL, 'clr')
physeq_clr_phylum <- microbiome::aggregate_taxa(physeq_clr, 'Phylum')
sample_metadata <- data.frame(physeq_clr@sam_data)%>%
  rownames_to_column(., "sample")
clr_t_phylum <- as.data.frame(t(as.matrix(physeq_clr_phylum@otu_table))) %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(1,3)], by = join_by(sample == sample))
significant_phylum <- clr_t_phylum %>%
    pivot_longer(-c(sample,Extraction), names_to = "Phylum", values_to = "value") %>%
    group_by(Phylum) %>%
    summarise(p_value = wilcox.test(value ~ Extraction)$p.value) %>%
    mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)
significant_phylum
```

# Alpha diversity
## Alpha diversity means by species

```{r alpha_div, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#Calculate Hill numbers
richness <- asv_table %>% 
            column_to_rownames(var="asv") %>% 
            dplyr::select(where(~!all(. == 0))) %>% 
            hilldiv(.,q=0) %>% 
            t() %>% 
            as.data.frame() %>%
            dplyr::rename(richness=1) %>%
            rownames_to_column(var="sample")

neutral <- asv_table %>% 
            column_to_rownames(var="asv") %>% 
            dplyr::select(where(~!all(. == 0))) %>% 
            hilldiv(.,q=1) %>% 
            t() %>% 
            as.data.frame() %>%
            dplyr::rename(neutral=1) %>%
            rownames_to_column(var="sample")
phylogenetic <- asv_table %>% 
            column_to_rownames(var="asv") %>% 
            dplyr::select(where(~!all(. == 0))) %>% 
            hilldiv(.,q=1,tree=tree) %>% 
            t() %>% 
            as.data.frame() %>%
            dplyr::rename(phylogenetic=1) %>%
            rownames_to_column(var="sample")
# Merge all metrics
alpha_div <- richness %>%
      full_join(neutral,by=join_by(sample==sample)) %>%
      full_join(phylogenetic,by=join_by(sample==sample))

```
```{r alpha_div_extraction_summary, comment="",echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
alpha_div %>%
  pivot_longer(-sample, names_to = "alpha", values_to = "value") %>%
  left_join(metadata_row, by = join_by(sample == sample)) %>%
    group_by(alpha)%>%
    summarise(
              Faecal_mean=mean(value[Extraction=="Faecal"], na.rm=T),
              Faecal_sd=sd(value[Extraction=="Faecal"], na.rm=T),
              Saliva_mean=mean(value[Extraction=="Saliva"], na.rm=T),
              Saliva_sd=sd(value[Extraction=="Saliva"], na.rm=T)) %>%
    mutate(
           Faecal=str_c(round(Faecal_mean,2),"±",round(Faecal_sd,2)),
           Saliva=str_c(round(Saliva_mean,2),"±",round(Saliva_sd,2))) %>% 
    arrange(-Faecal_mean) %>% 
    dplyr::select(alpha,Faecal,Saliva) %>% 
    tt()
```

## Plot diversities
```{r alpha_div_plot1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
alpha_div_pivot <- richness %>%
  full_join(neutral,by=join_by(sample==sample)) %>%
  full_join(phylogenetic,by=join_by(sample==sample)) %>%
  pivot_longer(-sample, names_to = "data", values_to = "value") %>%
  left_join(., metadata_row, by = join_by(sample == sample))

alpha_div_pivot %>%
  ggplot(aes(x=value, y=sample)) +
  geom_bar(stat='identity', fill="#6c9ebc") +
  facet_nested(Extraction ~ data,  scales="free")+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(strip.background = element_blank(),
    panel.grid.minor.x = element_line( size=.1, color="grey" ),
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.text.y = element_text(size=6))
```


```{r alpha_div_species, comment="", echo=FALSE, message=FALSE, warning=FALSE}
alpha_colors <- c("#e5bd5b", "#6b7398", "#76b183", "#d57d2c", "#2a2d26", "#f9d4cc", "#3c634e", "#ea68c3")

group_n <- alpha_div %>%
  left_join(., metadata_row, by = join_by(sample == sample))%>%
  select(Extraction) %>%
  pull() %>%
  unique() %>%
  length()
#pdf("figures/richness_div_body.pdf",width=7, height=6)
 alpha_div %>%
  left_join(., metadata_row, by = join_by(sample == sample))%>%
  ggplot(aes(x = Extraction, y = richness, group = Extraction, color = Extraction, fill = Extraction)) +
   geom_boxplot(alpha = 0.2,width = 0.5, outlier.shape = NA, show.legend = FALSE)+
   geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE) +
  scale_color_manual(values = alpha_colors[c(1:group_n)]) +
  scale_fill_manual(values = paste0(alpha_colors[c(1:group_n)], "50")) +
  theme(axis.text.x = element_text(vjust = 0.5, size = 10),
        axis.text.y = element_text(size = 10),
        axis.title = element_text(size = 14, face = "bold"),
        axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        strip.text.x = element_text(size = 12, colour = "black")) +
  guides(colour = guide_legend(override.aes = list(size = 6))) +
  labs(x = "Body site", y = "Richness")
 #dev.off()

#pdf("figures/neutral_div_body.pdf",width=7, height=6)
alpha_div %>%
  left_join(., metadata_row, by = join_by(sample == sample))%>%
  ggplot(aes(x = Extraction, y = neutral, group = Extraction, color = Extraction, fill = Extraction)) +
   geom_boxplot(alpha = 0.2, width = 0.5, outlier.shape = NA, show.legend = FALSE)+
   geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE) +
  scale_color_manual(values = alpha_colors[c(1:group_n)]) +
  scale_fill_manual(values = paste0(alpha_colors[c(1:group_n)], "50")) +
  theme(axis.text.x = element_text(vjust = 0.5, size = 10),
        axis.text.y = element_text(size = 10),
        axis.title = element_text(size = 14, face = "bold"),
        axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        strip.text.x = element_text(size = 12, colour = "black")) +
  guides(colour = guide_legend(override.aes = list(size = 6))) +
  labs(x = "Body site", y = "Neutral")
#dev.off()

#pdf("figures/phylo_div_body.pdf",width=7, height=6)
alpha_div %>%
  left_join(., metadata_row, by = join_by(sample == sample))%>%
  ggplot(aes(x = Extraction, y = phylogenetic, group = Extraction, color = Extraction, fill = Extraction)) +
   geom_boxplot(alpha = 0.2, width = 0.5, outlier.shape = NA, show.legend = FALSE)+
   geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE) +
  scale_color_manual(values = alpha_colors[c(1:group_n)]) +
  scale_fill_manual(values = paste0(alpha_colors[c(1:group_n)], "50")) +
  theme(axis.text.x = element_text(vjust = 0.5, size = 10),
        axis.text.y = element_text(size = 10),
        axis.title = element_text(size = 14, face = "bold"),
        axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        strip.text.x = element_text(size = 12, colour = "black")) +
  guides(colour = guide_legend(override.aes = list(size = 6))) +
  labs(x = "Body site", y = "Phylogenetic")
#dev.off()
```

## Mixed models
```{r}
set.seed(1411)
```

```{r}
alpha_div_meta <- alpha_div %>%
  left_join(., metadata_row, by = join_by(sample == sample))

#richness: Negative Binomial
Modelq0 <- glmer.nb(richness ~ Extraction+(1|DogID), data = alpha_div_meta)
summary(Modelq0)
r.squaredGLMM(Modelq0)
#Plot the model
plot_model(Modelq0,type="pred",terms = "Extraction", show.data = TRUE,jitter=0.2, connect.lines = TRUE)+theme_bw()

#q1:linear
Modelq1 <- lme(fixed = neutral ~ Extraction, data = alpha_div_meta,
               random = ~ 1 | DogID)
summary(Modelq1)
anova(Modelq1)
plot_model(Modelq1,type="pred",terms = "Extraction", show.data = TRUE,jitter=0.2, connect.lines = TRUE)+theme_bw()

#q1 phylogenetic
Modelq1p <- lme(fixed = phylogenetic ~ Extraction, data = alpha_div_meta,
               random = ~ 1 | DogID)
summary(Modelq1p)
anova(Modelq1p)
```

# Beta diversity
```{r beta_div, comment="", echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
beta_q0 <- asv_table %>%
  column_to_rownames(., "asv")%>%
  hillpair(., q = 0)

beta_q1n <- asv_table %>%
  column_to_rownames(., "asv")%>%
  hillpair(., q = 1)

beta_q1p <- asv_table%>%
  column_to_rownames(., "asv")%>%
  hillpair(., q = 1, tree = tree)

save(beta_q0,beta_q1n,beta_q1p, file = "resources/beta_div_FL_filtered.Rdata")
```

```{r}
load("resources/beta_div_FL_filtered.Rdata")
```

```{r betan, comment="", echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
beta_metric <- beta_q0n$S

beta_q0n_nmds <- beta_metric %>%
  vegan::metaMDS(., trymax = 500, k = 2, verbosity = FALSE) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(metadata_row, by = join_by(sample == sample))

group_n <- length(unique(beta_q0n_nmds$Extraction))
beta_colors <- c("#e5bd5b", "#6b7398", "#76b183", "#d57d2c", "#2a2d26", "#f9d4cc", "#3c634e", "#ea68c3")
```

## Neutral beta diversity
```{r betan, comment="", echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
beta_metric <- beta_q1n$S

beta_q1n_nmds <- beta_metric %>%
  vegan::metaMDS(., trymax = 500, k = 2, verbosity = FALSE) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(metadata_row, by = join_by(sample == sample))

group_n <- length(unique(beta_q1n_nmds$Extraction))
beta_colors <- c("#e5bd5b", "#6b7398", "#76b183", "#d57d2c", "#2a2d26", "#f9d4cc", "#3c634e", "#ea68c3")
```
```{r beta_div_neutral, comment="", echo=FALSE, message=FALSE, warning=FALSE}
q1n_nmds <- beta_q1n_nmds %>%
  group_by(Extraction) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup()

#pdf("figures/beta_body_sites.pdf",width=14, height=9)
ggplot(q1n_nmds, aes(x = NMDS1, y = NMDS2, color = Extraction)) +
  scale_color_manual(values = beta_colors[c(1:group_n)]) +
  geom_point(size = 2) +
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
  theme_classic() +
  theme(
    axis.text.x = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.title = element_text(size = 14, face = "bold"),
    axis.text = element_text(face = "bold", size = 12),
    panel.background = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 12),
    legend.title = element_text(size = 14),
    legend.position = "right", legend.box = "vertical"
  ) + 
  labs(colour = "Body site")
#dev.off()
```

***Homogeneity of variance***

```{r permu1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ps.disper.neutral <- betadisper(beta_metric, metadata_row$Extraction) 
permutest(ps.disper.neutral, pairwise = TRUE) 
```

***Permanova***
```{r adonisbeta2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#### Adonis
metarow <- column_to_rownames(metadata_row, "sample")

adonis2(beta_metric ~ Extraction, data = metarow[labels(beta_metric), ], permutations = 999) %>%
  as.matrix() %>%
  knitr::kable(., digits = c(0,3,3,3,3))
```

## Phylogenetic beta diversity
```{r betap, comment="", echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
betap_metric <- beta_q1p$S

beta_q1p_nmds <- betap_metric %>%
  vegan::metaMDS(., trymax = 500, k = 2, verbosity = FALSE) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(metadata_row, by = join_by(sample == sample))

group_n <- length(unique(beta_q1p_nmds$Extraction))
beta_colors <- c("#e5bd5b", "#6b7398", "#76b183", "#d57d2c", "#2a2d26", "#f9d4cc", "#3c634e", "#ea68c3")
```
```{r beta_div_phylo, comment="", echo=FALSE, message=FALSE, warning=FALSE}
q1p_nmds <- beta_q1p_nmds %>%
  group_by(Extraction) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup()
ggplot(q1p_nmds, aes(x = NMDS1, y = NMDS2, color = Extraction)) +
  scale_color_manual(values = beta_colors[c(1:group_n)]) +
  scale_shape_manual(values = 1:10) +
  geom_point(size = 2) +
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9) +
  theme_classic() +
  theme(
    axis.text.x = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.title = element_text(size = 14, face = "bold"),
    axis.text = element_text(face = "bold", size = 12),
    panel.background = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 12),
    legend.title = element_text(size = 14),
    legend.position = "right", legend.box = "vertical"
  )
```

***Homogeneity of variance***
```{r permup, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ps.disper.neutral <- betadisper(betap_metric, metadata_row$Extraction) 
permutest(ps.disper.neutral, pairwise = TRUE) 
```

***Permanova***
```{r adonisbetap, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#### Adonis
metarow <- column_to_rownames(metadata_row, "sample")

adonis2(betap_metric ~ Extraction, data = metarow[labels(beta_metric), ], permutations = 999) %>%
  as.matrix() %>%
  knitr::kable(., digits = c(0,3,3,3,3))
```

