---
title: "Sled_dog_manuscript"
output:
  pdf_document: default
  html_document: default
date: "2023-03-03"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Data preparation

## Load required libraries

```{r libraries}
library(vegan)
library(phyloseq)
library(hilldiv)
library(ape)
library(phytools)
library(dplyr)
library(tibble)
library(microbiome)
library(ggplot2)
library(RColorBrewer)
library(ggpubr)
library(ranacapa)
library(phylox)
library(lme4)
library(MuMIn)
library(sjPlot)
library(nlme)
library(matrixStats)
library(microbiomeutilities)
library(emmeans)
```

# Saliva samples
```{r}
physeq_phylum_clean <- readRDS("resources/physeq_phylum_filtered_tree.RData")
physeq_phylum_clean <- prune_taxa(taxa_sums(physeq_phylum_clean)>0, physeq_phylum_clean)
physeq_saliva <- subset_samples(physeq_phylum_clean, Extraction == "Saliva")
physeq_saliva <- subset_samples(physeq_saliva, Working_status %in% c("Follower", "Lead"))
physeq_saliva <- prune_taxa(taxa_sums(physeq_saliva)>0, physeq_saliva)
```

```{r saliva}
saliva_table <- data.frame(physeq_saliva@otu_table)
saliva_metatable <- data.frame(physeq_saliva@sam_data)

saliva_taxonomy <- data.frame(physeq_saliva@tax_table)%>% rownames_to_column(., "asv")

tree_saliva = phy_tree(physeq_saliva)
tree_saliva <- force.ultrametric(tree_saliva, method = "extend")

saliva_table <- data.frame(physeq_saliva@otu_table)
identical(sort(colnames(saliva_table)),sort(as.character(saliva_metatable[,1])))
match_data(saliva_table,tree_saliva)
saliva_table <- data.frame(physeq_saliva@otu_table)%>% rownames_to_column(., "asv")

saliva_metatable$Pack=factor(saliva_metatable$Pack)
saliva_metatable$Group <- paste("Group", saliva_metatable$Pack, sep = "_")
sample_data(physeq_saliva) <- saliva_metatable
saliva_metatable<- saliva_metatable %>% rownames_to_column(., "sample")
```

## Explore the number of samples 
```{r samples_saliva}
table(meta(physeq_saliva)$Sex, meta(physeq_saliva)$Location)
table(meta(physeq_saliva)$Working_status, meta(physeq_saliva)$Location)
```
## Bacteria with phylum information
```{r bact_class, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_class <- subset_taxa(physeq_saliva, !is.na(Class))
physeq_class <- prune_taxa(taxa_sums(physeq_class)>0, physeq_class)
cat(nrow(physeq_class@tax_table))
```
```{r bact_class_info, comment="", echo=FALSE, message=FALSE, warning=FALSE}
table(tax_table(physeq_class)[,"Class"]) %>% as.data.frame() %>% arrange(desc(Freq))
```

## Number of unique Phyla
```{r bact_phyla, comment="", echo=FALSE, message=FALSE, warning=FALSE}
phyla <- get_taxa_unique(physeq_saliva, "Phylum")
cat(length(phyla))
```

## Bacteria without class information
```{r bact_noclass, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_classNA <- subset_taxa(physeq_saliva, is.na(Class))
cat(nrow(physeq_classNA@tax_table))
```
```{r bact_class_info, comment="", echo=FALSE, message=FALSE, warning=FALSE}
table(tax_table(physeq_classNA)[,"Phylum"]) %>% as.data.frame() %>% arrange(desc(Freq))
```

## Number of unique orders
```{r bact_order, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#get_taxa_unique(physeq_class, "Order")
#table(tax_table(physeq_class)[,"Order"]) %>% as.data.frame()
order <- get_taxa_unique(physeq_class, "Order")
cat(length(order))
```

## Number of unique families
```{r bact_fam, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#get_taxa_unique(physeq_class, "Family")
family <- get_taxa_unique(physeq_class, "Family")
cat(length(family))
```

## Number of unique genera
```{r bact_gene, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genus <- get_taxa_unique(physeq_class, "Genus")
cat(length(genus))
```

## Barplots by working status
```{r barplot_na, comment="", echo=FALSE, message=FALSE, warning=FALSE}
metadata_row <- data.frame(sample_data(physeq_saliva)) %>% rownames_to_column(., "sample")
asv_table<- data.frame(physeq_saliva@otu_table)%>% rownames_to_column(., "asv")
taxonomy_table <- data.frame(physeq_saliva@tax_table)%>% rownames_to_column(., "asv")

asv_table %>%
  mutate_at(vars(-asv),~./sum(.)) %>% #apply TSS normalisation
  pivot_longer(-asv, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., taxonomy_table, by = join_by(asv == asv)) %>% #append genome metadata
  left_join(., metadata_row, by = join_by(sample == sample)) %>% #append sample metadata
  ggplot(., aes(x=sample,y=count, fill=Phylum, group=Phylum)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) +
  facet_grid(.~Working_status,  scales="free_x")+
theme(axis.text.x=element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x=element_blank(),
      axis.title = element_text(size = 10, face = "bold"),
    panel.background = element_blank(),
    # panel.border = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 6),
    legend.title = element_text(size=10, face = "bold"),
    legend.key.size = unit(0.3, 'cm')) +
  labs(x = "Sample", y = "Relative abundances",fill="Phylum")+
    guides(fill = guide_legend(ncol = 1)) 
```

## Barplots by working status
```{r barplot_location, comment="", echo=FALSE, message=FALSE, warning=FALSE}
top_phylum <- aggregate_top_taxa2(physeq_saliva, top = 10, "Phylum")
metadata_row_top_phylum <- data.frame(sample_data(top_phylum)) %>% rownames_to_column(., "sample")
asv_table_top_phylum<- data.frame(top_phylum@otu_table)%>% rownames_to_column(., "asv")
taxonomy_table_top_phylum <- data.frame(top_phylum@tax_table)%>% rownames_to_column(., "asv")

asv_table_top_phylum %>%
  mutate_at(vars(-asv),~./sum(.)) %>% 
  pivot_longer(-asv, names_to = "sample", values_to = "count") %>% 
  left_join(., taxonomy_table_top_phylum, by = join_by(asv == asv)) %>% 
  left_join(., metadata_row_top_phylum, by = join_by(sample == sample)) %>% 
  ggplot(., aes(x=sample,y=count, fill=Phylum, group=Phylum)) +
  geom_bar(stat="identity", colour="white", linewidth=0.1) +
  facet_grid(.~Location,  scales="free_x")+
  theme(axis.text.x=element_blank(),
        axis.title.x = element_blank(),
        axis.ticks.x=element_blank(),
        axis.title = element_text(size = 10, face = "bold"),
    panel.background = element_blank(),
    # panel.border = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 6),
    legend.title = element_text(size=10, face = "bold"),
    legend.key.size = unit(0.3, 'cm')) +
  labs(x = "Sample", y = "Relative abundances",fill="Phylum")+
    guides(fill = guide_legend(ncol = 1)) 
```

## Phylum
```{r phylum_saliva}
physeq_saliva.phylum <- microbiome::aggregate_taxa(physeq_saliva, 'Phylum')
physeq_saliva.phylum.rel <-  microbiome::transform(physeq_saliva.phylum, "compositional")
saliva.rel <- physeq_saliva.phylum.rel@otu_table*100
means.saliva.rel <- as.data.frame(rowMeans(saliva.rel))
sd.saliva.rel <- as.data.frame(rowSds(saliva.rel, useNames = TRUE))
summary.phylum.saliva <- merge(means.saliva.rel, sd.saliva.rel, by="row.names")
colnames(summary.phylum.saliva) <- c("Phylum","mean", "sd")
summary.phylum.saliva[order(-summary.phylum.saliva$mean),]
```

## Order
```{r order_saliva}
physeq_saliva.order <- microbiome::aggregate_taxa(physeq_saliva, 'Order')
physeq_saliva.order.rel <-  microbiome::transform(physeq_saliva.order, "compositional")
saliva.order.rel <- physeq_saliva.order.rel@otu_table*100
means.saliva.order.rel <- as.data.frame(rowMeans(saliva.order.rel))
sd.saliva.order.rel <- as.data.frame(rowSds(saliva.order.rel, useNames = TRUE))
summary.order.saliva <- merge(means.saliva.order.rel, sd.saliva.order.rel, by="row.names")
colnames(summary.order.saliva) <- c("order","mean", "sd")
percentage.saliva.order.sort <- summary.order.saliva[order(-summary.order.saliva$mean),]
percentage.saliva.order.sort[1-10,]
```

## Family
```{r family_saliva}
physeq_saliva.family <- microbiome::aggregate_taxa(physeq_saliva, 'Family')
physeq_saliva.family.rel <-  microbiome::transform(physeq_saliva.family, "compositional")
saliva.family.rel <- physeq_saliva.family.rel@otu_table*100
means.saliva.family.rel <- as.data.frame(rowMeans(saliva.family.rel))
sd.saliva.family.rel <- as.data.frame(rowSds(saliva.family.rel, useNames = TRUE))
summary.family.saliva <- merge(means.saliva.family.rel, sd.saliva.family.rel, by="row.names")
colnames(summary.family.saliva) <- c("family","mean", "sd")
percentage.saliva.family.sort <- summary.family.saliva[order(-summary.family.saliva$mean),]
percentage.saliva.family.sort[1-10,]
```

## Genus
```{r genus_saliva}
physeq_saliva.genus <- microbiome::aggregate_taxa(physeq_saliva, 'Genus')
physeq_saliva.genus.rel <-  microbiome::transform(physeq_saliva.genus, "compositional")
saliva.genus.rel <- physeq_saliva.genus.rel@otu_table*100
means.saliva.genus.rel <- as.data.frame(rowMeans(saliva.genus.rel))
sd.saliva.genus.rel <- as.data.frame(rowSds(saliva.genus.rel, useNames = TRUE))
summary.genus.saliva <- merge(means.saliva.genus.rel, sd.saliva.genus.rel, by="row.names")
colnames(summary.genus.saliva) <- c("family","mean", "sd")
percentage.saliva.genus.sort <- summary.genus.saliva[order(-summary.genus.saliva$mean),]
percentage.saliva.genus.sort[1-10,]
```

# Alpha diversity
```{r alpha_div, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#Calculate Hill numbers
richness <- saliva_table %>% 
            column_to_rownames(var="asv") %>% 
            dplyr::select(where(~!all(. == 0))) %>% 
            hilldiv(.,q=0) %>% 
            t() %>% 
            as.data.frame() %>%
            dplyr::rename(richness=1) %>%
            rownames_to_column(var="sample")

neutral <- saliva_table %>% 
            column_to_rownames(var="asv") %>% 
            dplyr::select(where(~!all(. == 0))) %>% 
            hilldiv(.,q=1) %>% 
            t() %>% 
            as.data.frame() %>%
            dplyr::rename(neutral=1) %>%
            rownames_to_column(var="sample")
phylogenetic <- saliva_table %>% 
            column_to_rownames(var="asv") %>% 
            dplyr::select(where(~!all(. == 0))) %>% 
            hilldiv(.,q=1,tree=tree_saliva) %>% 
            t() %>% 
            as.data.frame() %>%
            dplyr::rename(phylogenetic=1) %>%
            rownames_to_column(var="sample")
# Merge all metrics
alpha_div <- richness %>%
      full_join(neutral,by=join_by(sample==sample)) %>%
      full_join(phylogenetic,by=join_by(sample==sample)) %>%
      left_join(., saliva_metatable, by = join_by(sample == sample))


richness_mean <- alpha_div %>%
  group_by(Pack, Working_status) %>%
  dplyr::summarise_at(.vars = names(.)[2], .funs = c("Richness mean" = "mean", "Richness sd" = "sd"))

neutral_mean <- alpha_div %>%
  group_by(Pack, Working_status) %>%
  dplyr::summarise_at(.vars = names(.)[3], .funs = c("Neutral mean" = "mean", "Neutral sd" = "sd"))

phylogenetic_mean <- alpha_div %>%
  group_by(Pack, Working_status) %>%
  dplyr::summarise_at(.vars = names(.)[4], .funs = c("Phylogenetic mean" = "mean", "Phylogenetic sd" = "sd"))


cbind(richness_mean, neutral_mean[, 3:4], phylogenetic_mean[, 3:4])

```

## Plot diversities
```{r alpha_div_plot1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
alpha_div_pivot <- richness %>%
  full_join(neutral,by=join_by(sample==sample)) %>%
  full_join(phylogenetic,by=join_by(sample==sample)) %>%
  pivot_longer(-sample, names_to = "data", values_to = "value") %>%
  left_join(., saliva_metatable, by = join_by(sample == sample))

alpha_div_pivot %>%
  ggplot(aes(x=value, y=sample)) +
  geom_bar(stat='identity', fill="#6c9ebc") +
  facet_nested(Pack ~ data,  scales="free")+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(strip.background = element_blank(),
    panel.grid.minor.x = element_line( size=.1, color="grey" ),
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.text.y = element_text(size=6))
```

```{r alpha_div_species, comment="", echo=FALSE, message=FALSE, warning=FALSE}
alpha_colors <- c("#e5bd5b", "#6b7398", "#76b183", "#d57d2c", "#2a2d26", "#f9d4cc", "#3c634e", "#ea68c3")
group_n <- alpha_div %>%
  select(Pack) %>%
  pull() %>%
  unique() %>%
  length()

highlight_df <- alpha_div %>% 
             filter(Working_status=="Lead")

 alpha_div %>%
  ggplot(aes(x = Pack, y = richness, group = Pack, color = Pack, fill = Pack)) +
   geom_boxplot(alpha = 0.2, outlier.shape = NA, show.legend = FALSE)+
   geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE) + 
  geom_point(data=highlight_df, 
             aes(x=Pack,y=richness), 
             color='red',
             size=2, show.legend = FALSE)+
  scale_color_manual(values = alpha_colors[c(1:group_n)]) +
  scale_fill_manual(values = paste0(alpha_colors[c(1:group_n)], "50")) +
  theme(axis.text.x = element_text(vjust = 0.5, size = 10),
        axis.text.y = element_text(size = 10),
        axis.title = element_text(size = 14, face = "bold"),
        axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        strip.text.x = element_text(size = 12, colour = "black")) +
  guides(colour = guide_legend(override.aes = list(size = 6))) +
  labs(x = "Pack", y = "Richness")


```


```{r alpha_div_neutral, comment="", echo=FALSE, message=FALSE, warning=FALSE}
alpha_colors <- c("#e5bd5b", "#6b7398", "#76b183", "#d57d2c", "#2a2d26", "#f9d4cc", "#3c634e", "#ea68c3")
group_n <- alpha_div %>%
  select(Pack) %>%
  pull() %>%
  unique() %>%
  length()
#pdf("figures/neutral_saliva_bigger.pdf",width=14, height=9)
 alpha_div %>%
  ggplot(aes(x = Group, y = neutral, group = Group, color = Group, fill = Group)) +
   geom_boxplot(alpha = 0.2, outlier.shape = NA, show.legend = FALSE)+
   geom_jitter(width = 0.2, size = 2.5, show.legend = FALSE) + 
  geom_point(data=highlight_df, 
             aes(x=Group,y=neutral), 
             color='red',
             size=4, show.legend = FALSE)+
  scale_color_manual(values = alpha_colors[c(1:group_n)]) +
  scale_fill_manual(values = paste0(alpha_colors[c(1:group_n)], "50")) +
  theme(axis.text.x = element_text(vjust = 0.5, size = 14),
        axis.text.y = element_text(size = 14),
        axis.title = element_text(size = 16, face = "bold"),
        axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.title = element_text(size=22, face="bold"),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black")) +
  guides(colour = guide_legend(override.aes = list(size = 6))) +
  labs(x = "Sled dog group", y = "Neutral alpha diversity", title="Saliva samples")
# dev.off()

```
```{r alpha_div_phylo, comment="", echo=FALSE, message=FALSE, warning=FALSE}
alpha_colors <- c("#e5bd5b", "#6b7398", "#76b183", "#d57d2c", "#2a2d26", "#f9d4cc", "#3c634e", "#ea68c3")
group_n <- alpha_div %>%
  select(Group) %>%
  pull() %>%
  unique() %>%
  length()

highlight_df <- alpha_div %>% 
             filter(Working_status=="Lead")
 alpha_div %>%
  ggplot(aes(x = Group, y = phylogenetic, group = Group, color = Group, fill = Group)) +
   geom_boxplot(alpha = 0.2, outlier.shape = NA, show.legend = FALSE)+
   geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE) + 
  geom_point(data=highlight_df, 
             aes(x=Group,y=phylogenetic), 
             color='red',
             size=2, show.legend = FALSE)+
  scale_color_manual(values = alpha_colors[c(1:group_n)]) +
  scale_fill_manual(values = paste0(alpha_colors[c(1:group_n)], "50")) +
  theme(axis.text.x = element_text(vjust = 0.5, size = 10),
        axis.text.y = element_text(size = 10),
        axis.title = element_text(size = 14, face = "bold"),
        axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        strip.text.x = element_text(size = 12, colour = "black")) +
  guides(colour = guide_legend(override.aes = list(size = 6))) +
  labs(x = "Group", y = "Richness")
 
```

```{r model_neutral, comment="", echo=FALSE, message=FALSE, warning=FALSE}
Model_neutral <- lm(formula = neutral ~ Working_status+Group, data = alpha_div)
summary(Model_neutral)
anova(Model_neutral)
emmeans(Model_neutral, pairwise ~ Group)

Model_neutral1 <- lme(fixed = neutral ~ Working_status, data = alpha_div,
               random = ~ 1 | Pack)
summary(Model_neutral1)
```
```{r model_phylo, comment="", echo=FALSE, message=FALSE, warning=FALSE}
Model_phylogenetic <- lm(formula = phylogenetic ~ Working_status+Group, data = alpha_div)
summary(Model_phylogenetic)
anova(Model_phylogenetic)
emmeans(Model_phylogenetic, pairwise ~ Group)

Model_phylogenetic1 <- lme(fixed = phylogenetic ~ Working_status, data = alpha_div,
               random = ~ 1 | Pack)
summary(Model_phylogenetic1)
```
# Beta diversity
```{r beta_div, comment="", echo=FALSE, message=FALSE, warning=FALSE, results='hide', eval=FALSE}
beta_q0 <- saliva_table %>%
  column_to_rownames(., "asv")%>%
  hillpair(., q = 0)

beta_q1n <- saliva_table %>%
  column_to_rownames(., "asv")%>%
  hillpair(., q = 1)

beta_q1p <- saliva_table%>%
  column_to_rownames(., "asv")%>%
  hillpair(., q = 1, tree = tree_faecal)

save(beta_q0,beta_q1n,beta_q1p, file = "resources/beta_div_saliva_filter.Rdata")
```

```{r load_beta, comment="", echo=FALSE, message=FALSE, warning=FALSE}
load("resources/beta_div_saliva_filter.Rdata")
```

## Neutral beta diversity
```{r}
set.seed(1411)
```

```{r betan, comment="", echo=FALSE, message=FALSE, warning=FALSE, results='hide'}

beta_metric <- beta_q1n$S

beta_q1n_nmds <- beta_metric %>%
  vegan::metaMDS(., trymax = 500, k = 2, verbosity = FALSE) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(saliva_metatable, by = join_by(sample == sample))

group_n <- length(unique(beta_q1n_nmds$Group))
beta_colors <- c("#e5bd5b", "#6b7398", "#76b183", "#d57d2c", "#2a2d26", "#f9d4cc", "#3c634e", "#ea68c3")
```

```{r beta_div_neutral, comment="", echo=FALSE, message=FALSE, warning=FALSE}
q1n_nmds <- beta_q1n_nmds %>%
  group_by(Group) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup()

#pdf("figures/neutral_beta_saliva.pdf",width=14, height=9)
ggplot(q1n_nmds, aes(x = NMDS1, y = NMDS2, color = Group)) +
  scale_color_manual(values = beta_colors[c(1:group_n)]) +
  scale_shape_manual(values = 1:10) +
  geom_point(size = 2) +
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
  theme_classic() +
  theme(
    axis.text.x = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.title = element_text(size = 14, face = "bold"),
    axis.text = element_text(face = "bold", size = 12),
    panel.background = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 12),
    legend.title = element_text(size = 14),
    legend.position = "right", legend.box = "vertical"
  ) + labs(color='Role on the group')
#dev.off()
```

***Homogeneity of variance***

```{r permu1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ps.disper.neutral <- betadisper(beta_metric, saliva_metatable$Group) 
permutest(ps.disper.neutral, pairwise = TRUE) 
```

***Permanova***
```{r adonisbeta2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#### Adonis
metarow <- column_to_rownames(saliva_metatable, "sample")

adonis2(beta_metric ~ Group+Working_status+Sex, data = metarow[labels(beta_metric), ], permutations = 999) %>%
  as.matrix() %>%
  knitr::kable(., digits = c(0,3,3,3,3))

adonis2(beta_metric ~ Working_status, data = metarow[labels(beta_metric), ], permutations = 999, strata =metarow$Group) %>%
  as.matrix() %>%
  knitr::kable(., digits = c(0,4,4,4,4,4))
```

