---
title: "Sled dogs: faecal samples"
output:
  pdf_document: default
  html_document: default
date: "2023-03-08"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Data preparation

<!-- This data has been generated from sled dog samples processed by Mads in the lab and Ostaizka doing the analyses -->

## Load required libraries

```{r libraries, message=FALSE, warning=FALSE, echo=FALSE}
library(vegan)
library(phyloseq)
library(hilldiv)
library(ape)
library(phytools)
library(dplyr)
library(tidyr)
library(tibble)
library(tidyverse)
library(microbiome)
library(ggplot2)
library("RColorBrewer")
library(ggpubr)
library(ranacapa)
library(phylox)
library(lme4)
library(MuMIn)
library(sjPlot)
library(nlme)
library(matrixStats)
library(microbiomeutilities)
library(ggh4x)
library(hilldiv2)
library(janitor)
```

## Declare directories and files

```{r directories, echo=FALSE}
physeq_phylum_clean <- readRDS("resources/physeq_phylum_filtered_tree.RData")
physeq_phylum_clean <- prune_taxa(taxa_sums(physeq_phylum_clean)>0, physeq_phylum_clean)
set.seed(2345)
```

# Filtering followers and leaders
```{r FL, echo=FALSE}
physeq_FL <- subset_samples(physeq_phylum_clean, Working_status %in% c("Follower", "Lead"))
physeq_FL <- prune_taxa(taxa_sums(physeq_FL)>0, physeq_FL)
```

# Filtering faecal samples
```{r faecal, warning=FALSE, echo=FALSE}
#physeq_faecal <- subset_samples(physeq_FL, Extraction == "Faecal")
#physeq_faecal = filter_taxa(physeq_faecal.rel, function(x) sum(x > 0.0005) > (0.005*length(x)), TRUE)
#physeq_faecal <- prune_taxa(taxa_sums(physeq_faecal)>0, physeq_faecal)
faecal_table <- data.frame(physeq_faecal@otu_table)
faecal_metadata <- data.frame(physeq_faecal@sam_data)
faecal_metadata$Pack=factor(faecal_metadata$Pack)
faecal_metadata$Group <- paste("Group", faecal_metadata$Pack, sep = "_")
#faecal_metadata <- column_to_rownames(faecal_metadata, "sample")
sample_data(physeq_faecal) <- faecal_metadata
tree_faecal = phy_tree(physeq_faecal)
hierarchy_faecal <- tibble::rownames_to_column(faecal_metadata, "Sample")
hierarchy_faecal <- hierarchy_faecal[which(hierarchy_faecal[,1] %in% colnames(faecal_table)),]
saveRDS(physeq_faecal,"resources/physeq_faecal.RData")
```

```{r faecal_match, echo=FALSE}
identical(sort(colnames(faecal_table)),sort(as.character(hierarchy_faecal[,1])))
match_data(faecal_table,tree_faecal)
```

# Explore the number of samples 
```{r samples_faecal, comment="", echo=FALSE}
table(meta(physeq_faecal)$Sex, meta(physeq_faecal)$Location)
table(meta(physeq_faecal)$Working_status, meta(physeq_faecal)$Location)
```

## Bacteria with phylum information
```{r bact_class, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_phylum <- subset_taxa(physeq_faecal, !is.na(Phylum))
physeq_phylum <- prune_taxa(taxa_sums(physeq_phylum)>0, physeq_phylum)
cat(nrow(physeq_phylum@tax_table))
```

## Number of unique Phyla
```{r bact_phyla, comment="", echo=FALSE, message=FALSE, warning=FALSE}
phyla <- get_taxa_unique(physeq_faecal, "Phylum")
cat(length(phyla))
```

## Bacteria with class information
```{r bact_class, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_class <- subset_taxa(physeq_faecal, !is.na(Class))
physeq_class <- prune_taxa(taxa_sums(physeq_class)>0, physeq_class)
cat(nrow(physeq_class@tax_table))
```
```{r bact_class_info, comment="", echo=FALSE, message=FALSE, warning=FALSE}
table(tax_table(physeq_class)[,"Class"]) %>% as.data.frame() %>% arrange(desc(Freq))
```

## Bacteria without class information
```{r bact_noclass, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_classNA <- subset_taxa(physeq_faecal, is.na(Class))
cat(nrow(physeq_classNA@tax_table))
```
```{r bact_class_info, comment="", echo=FALSE, message=FALSE, warning=FALSE}
table(tax_table(physeq_classNA)[,"Phylum"]) %>% as.data.frame() %>% arrange(desc(Freq))
```

## Number of unique orders
```{r bact_order, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#get_taxa_unique(physeq_class, "Order")
#table(tax_table(physeq_class)[,"Order"]) %>% as.data.frame()
order <- get_taxa_unique(physeq_class, "Order")
cat(length(order))
```

## Number of unique families
```{r bact_fam, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#get_taxa_unique(physeq_class, "Family")
family <- get_taxa_unique(physeq_class, "Family")
cat(length(family))
```

## Number of unique genera
```{r bact_gene, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genus <- get_taxa_unique(physeq_class, "Genus")
cat(length(genus))
```

## Bacteria without genera information
```{r bact_noclass, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_genussNA <- subset_taxa(physeq_faecal, is.na(Genus))
cat(nrow(physeq_genussNA@tax_table))
```

## Barplots by working status
```{r barplot_na, comment="", echo=FALSE, message=FALSE, warning=FALSE}
faecal_metadata <- data.frame(sample_data(physeq_faecal)) %>% rownames_to_column(., "sample")
faecal_table <- data.frame(physeq_faecal@otu_table)%>% rownames_to_column(., "asv")
faecal_taxonomy <- data.frame(physeq_faecal@tax_table)%>% rownames_to_column(., "asv")
sample_data(faecal_metadata)
faecal_table %>%
  mutate_at(vars(-asv),~./sum(.)) %>% #apply TSS normalisation
  pivot_longer(-asv, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., faecal_taxonomy, by = join_by(asv == asv)) %>% #append genome metadata
  left_join(., faecal_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  ggplot(., aes(x=sample,y=count, fill=Phylum, group=Phylum)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) +
#    scale_fill_manual(values=phylum_colors) +
  facet_grid(.~Group,  scales="free_x")+
#   facet_nested(.~Group+Working_status,  scales="free_x") +
theme(axis.text.x=element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x=element_blank(),
      axis.title = element_text(size = 10, face = "bold"),
      strip.text.x = element_text(size = 12, color="black"),
      strip.background = element_rect(colour=NA, fill=NA),
    panel.background = element_blank(),
    # panel.border = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 6),
    legend.title = element_text(size=10, face = "bold"),
    legend.key.size = unit(0.3, 'cm')) +
  labs(x = "Sample", y = "Relative abundances",fill="Phylum")+
    guides(fill = guide_legend(ncol = 1)) 
```

```{r}
top_phylum <- aggregate_top_taxa2(physeq_faecal, top = 10, "Phylum")
metadata_row_top_phylum <- data.frame(sample_data(top_phylum)) %>% rownames_to_column(., "sample")
asv_table_top_phylum<- data.frame(top_phylum@otu_table)%>% rownames_to_column(., "asv")
taxonomy_table_top_phylum <- data.frame(top_phylum@tax_table)%>% rownames_to_column(., "asv")

asv_table_top_meta <- asv_table_top_phylum %>%
  mutate_at(vars(-asv),~./sum(.)) %>% #apply TSS normalisation
  pivot_longer(-asv, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., taxonomy_table_top_phylum, by = join_by(asv == asv)) %>% #append genome metadata
  left_join(., metadata_row_top_phylum, by = join_by(sample == sample)) #append sample metadata

metadata_sorted <- metadata_row_top_phylum %>% arrange(Group, sample)
a <- ifelse(metadata_sorted$Working_status == "Lead", "red", "blue")

asv_table_top_meta %>%
  ggplot(., aes(x=sample,y=count, fill=Phylum, group=Phylum)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) +
#    scale_fill_manual(values=phylum_colors) +
#  facet_grid(.~Working_status,  scales="free_x")+
   facet_nested(.~Group+Working_status,  scales="free_x") +
theme(
      strip.text.x = element_text(size = 8, color="black"),
      strip.background = element_rect(colour=NA, fill=NA),
      axis.text.x = element_blank(), #, color=a
      axis.ticks = element_blank(),
      axis.title = element_text(size = 10, face = "bold"),
    panel.background = element_blank(),
    # panel.border = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 6),
    legend.title = element_text(size=10, face = "bold"),
    legend.key.size = unit(0.3, 'cm')) +
  labs(x = "Sample", y = "Relative abundances",fill="Top 10 phyla")+
    guides(fill = guide_legend(ncol = 1)) 
```

```{r alpha_div, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#Calculate Hill numbers
richness <- faecal_table %>% 
#            column_to_rownames(var="asv") %>% 
            dplyr::select(where(~!all(. == 0))) %>% 
            hilldiv(.,q=0) %>% 
            t() %>% 
            as.data.frame() %>%
            dplyr::rename(richness=1) %>%
            rownames_to_column(var="sample")

neutral <- faecal_table %>% 
#            column_to_rownames(var="asv") %>% 
            dplyr::select(where(~!all(. == 0))) %>% 
            hilldiv(.,q=1) %>% 
            t() %>% 
            as.data.frame() %>%
            dplyr::rename(neutral=1) %>%
            rownames_to_column(var="sample")
phylogenetic <- faecal_table %>% 
#            column_to_rownames(var="asv") %>% 
            dplyr::select(where(~!all(. == 0))) %>% 
            hilldiv(.,q=1,tree=tree_faecal) %>% 
            t() %>% 
            as.data.frame() %>%
            dplyr::rename(phylogenetic=1) %>%
            rownames_to_column(var="sample")
# Merge all metrics
alpha_div <- richness %>%
      full_join(neutral,by=join_by(sample==sample)) %>%
      full_join(phylogenetic,by=join_by(sample==sample)) %>%
#      full_join(functional,by=join_by(sample==sample)) %>%
      left_join(., faecal_metadata, by = join_by(sample == sample))
#Pack
richness_mean <- alpha_div %>%
  group_by(Group) %>%
  dplyr::summarise_at(.vars = names(.)[2], .funs = c("Richness mean" = "mean", "Richness sd" = "sd"))

neutral_mean <- alpha_div %>%
  group_by(Group) %>%
  dplyr::summarise_at(.vars = names(.)[3], .funs = c("Neutral mean" = "mean", "Neutral sd" = "sd"))

phylogenetic_mean <- alpha_div %>%
  group_by(Group) %>%
  dplyr::summarise_at(.vars = names(.)[4], .funs = c("Phylogenetic mean" = "mean", "Phylogenetic sd" = "sd"))

cbind(richness_mean, neutral_mean[, 2:3], phylogenetic_mean[, 2:3])


```
```{r}
#Pack and working_status
richness_mean <- alpha_div %>%
  group_by(Group, Working_status) %>%
  dplyr::summarise_at(.vars = names(.)[2], .funs = c("Richness mean" = "mean", "Richness sd" = "sd"))

neutral_mean <- alpha_div %>%
  group_by(Group, Working_status) %>%
  dplyr::summarise_at(.vars = names(.)[3], .funs = c("Neutral mean" = "mean", "Neutral sd" = "sd"))

phylogenetic_mean <- alpha_div %>%
  group_by(Group, Working_status) %>%
  dplyr::summarise_at(.vars = names(.)[4], .funs = c("Phylogenetic mean" = "mean", "Phylogenetic sd" = "sd"))

cbind(richness_mean, neutral_mean[, 3:4], phylogenetic_mean[, 3:4])
```


## Plot diversities
```{r alpha_div_plot1, comment="", echo=FALSE, message=FALSE, warning=FALSE,fig.height=8, fig.width=10, fig.fullwidth=TRUE}
alpha_div_pivot <- richness %>%
  full_join(neutral,by=join_by(sample==sample)) %>%
  full_join(phylogenetic,by=join_by(sample==sample)) %>%
  pivot_longer(-sample, names_to = "data", values_to = "value") %>%
  left_join(., faecal_metadata, by = join_by(sample == sample))

alpha_div_pivot$data    <- factor( alpha_div_pivot$data, levels = c("richness", "neutral", "phylogenetic"))
alpha_div_pivot %>%
  ggplot(aes(x=value, y=sample)) +
  geom_bar(stat='identity', fill="#6c9ebc") +
  facet_nested(Group ~ data,  scales="free")+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(strip.background = element_blank(),
    panel.grid.minor.x = element_line( size=.1, color="grey" ),
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1, size = 6),
    axis.text.y = element_text(size=6))
```

```{r alpha_div_species, comment="", echo=FALSE, message=FALSE, warning=FALSE}
alpha_colors <- c("#e5bd5b", "#6b7398", "#76b183", "#d57d2c", "#2a2d26", "#f9d4cc", "#3c634e", "#ea68c3")
group_n <- alpha_div %>%
  select(Group) %>%
  pull() %>%
  unique() %>%
  length()

highlight_df <- alpha_div %>% 
             filter(Working_status=="Lead")
 alpha_div %>%
  ggplot(aes(x = Group, y = richness, group = Group, color = Group, fill = Group)) +
   geom_boxplot(alpha = 0.2, outlier.shape = NA, show.legend = FALSE)+
   geom_jitter(width = 0.2, size = 1.5, show.legend = FALSE) + 
  geom_point(data=highlight_df, 
             aes(x=Group,y=richness), 
             color='red',
             size=2, show.legend = FALSE)+
#  geom_violin(alpha = 0.2, width = 0.3, show.legend = FALSE) +
  scale_color_manual(values = alpha_colors[c(1:group_n)]) +
  scale_fill_manual(values = paste0(alpha_colors[c(1:group_n)], "50")) +
#  stat_compare_means(show.legend = FALSE) +
  theme(axis.text.x = element_text(vjust = 0.5, size = 10),
        axis.text.y = element_text(size = 10),
        axis.title = element_text(size = 14, face = "bold"),
        axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        strip.text.x = element_text(size = 12, colour = "black")) +
  guides(colour = guide_legend(override.aes = list(size = 6))) +
  labs(x = "Group", y = "Richness")
 
#pdf("figures/neutral_faecal_bigger.pdf",width=14, height=9)
 alpha_div %>%
  ggplot(aes(x = Group, y = neutral, group = Group, color = Group, fill = Group)) +
   geom_boxplot(alpha = 0.2, outlier.shape = NA, show.legend = FALSE)+
   geom_jitter(width = 0.2, size = 2.5, show.legend = FALSE) + 
  geom_point(data=highlight_df, 
             aes(x=Group,y=neutral), 
             color='red',
             size=4, show.legend = FALSE)+
#  geom_violin(alpha = 0.2, width = 0.3, show.legend = FALSE) +
  scale_color_manual(values = alpha_colors[c(1:group_n)]) +
  scale_fill_manual(values = paste0(alpha_colors[c(1:group_n)], "50")) +
#  stat_compare_means(show.legend = FALSE) +
  theme(axis.text.x = element_text(vjust = 0.5, size = 14),
        axis.text.y = element_text(size = 14),
        axis.title = element_text(size = 16, face = "bold"),
        axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.title = element_text(size=22, face="bold"),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black")) +
  guides(colour = guide_legend(override.aes = list(size = 6))) +
  labs(x = "Sled dog group", y = "Neutral alpha diversity", title="Faecal samples")
# dev.off()

```
## Mixed models

```{r}
library(lme4)
library(MuMIn)

##Negative Binomial
Modelq0GLMMNB <- glmer.nb(richness ~ Working_status+Group+(1|Group), data = alpha_div)#USE THIS!
summary(Modelq0GLMMNB)
r.squaredGLMM(Modelq0GLMMNB)

#Plot the model
library(sjPlot)
#pdf(paste(projectdirR,"Figures/GLMMNB_all_q0.pdf",sep=""),width=7, height=6)
plot_model(Modelq0GLMMNB,type="pred",terms = "Working_status", show.data = TRUE,jitter=0.2, connect.lines = TRUE)+theme_bw()
#dev.off()
#type = "pred"=Predicted values (marginal effects) for specific model terms. See ggpredict for details.

```
```{r}
#q1:linear
library(nlme)
Modelq1 <- lme(fixed = neutral ~ Working_status+Group, data = alpha_div,
               random = ~ 1 | Group)
summary(Modelq1)
anova(Modelq1)
AIC(Modelq1)# the first model is more appropriate, lower AIC
r.squaredGLMM(Modelq1)
Model_neutral <- lm(formula = neutral ~ Working_status+Group, data = alpha_div)
summary(Model_neutral)
anova(Model_neutral)
emmeans(Model_neutral, pairwise ~ Group)

#pdf(paste(projectdirR,"Figures/GLMM_all_q1.pdf",sep=""),width=7, height=6)
plot_model(Modelq1,type="pred",terms = "Working_status", show.data = TRUE,jitter=0.2, connect.lines = TRUE)+theme_bw()
#dev.off()
plot_model(Modelq1,type="re",terms = "Working_status", show.data = TRUE,jitter=0.2, connect.lines = TRUE)+theme_bw()
```

# Phylum percentages
```{r phylum_faecal, comment="", echo=FALSE}
physeq_faecal.phylum <- microbiome::aggregate_taxa(physeq_faecal, 'Phylum')
physeq_faecal.phylum.rel <-  microbiome::transform(physeq_faecal.phylum, "compositional")
faecal.rel <- physeq_faecal.phylum.rel@otu_table*100
means.faecal.rel <- as.data.frame(rowMeans(faecal.rel))
sd.faecal.rel <- as.data.frame(rowSds(faecal.rel, useNames = TRUE))
summary.phylum.faecal <- merge(means.faecal.rel, sd.faecal.rel, by="row.names")
colnames(summary.phylum.faecal) <- c("Phylum","mean", "sd")
print(summary.phylum.faecal[order(-summary.phylum.faecal$mean),], row.names = FALSE)
```
```{r phylum_faecal_plot, fig.cap="Plot stacked barplot colored by phylum", echo=FALSE}
top_phylum <- aggregate_top_taxa2(physeq_faecal.phylum.rel, top = 10, "Phylum")
dat.dataframe.top_phylum = psmelt(top_phylum)
ggplot(dat.dataframe.top_phylum, aes(x=Sample, y=Abundance, fill=OTU))+ 
  theme_light() + 
  geom_bar(stat="identity", position="stack") + 
  facet_grid(~Location, scale="free") + 
  theme(strip.text.x = element_text(size = 12, color="black"),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(),strip.background = element_rect(colour=NA, fill=NA),
        panel.border = element_rect(colour = "black", fill = NA), legend.text=element_text(size=6),
        axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),axis.text=element_text(size=6), axis.title=element_text(size=10,face="bold")) +
  scale_fill_brewer(palette = "Paired")
```

```{r}
faecal_phylum_rel <- physeq_faecal.phylum.rel@otu_table
phylum_mean <- faecal_phylum_rel %>%
  t() %>%
  as.data.frame()%>%
  rownames_to_column(., "sample") %>%
  left_join(., faecal_metadata_row, by = join_by(sample == sample)) %>%
  group_by(Working_status) %>%
  summarise_at(.vars = names(.)[2:8],.funs = c(mean="mean")) %>%
  t() %>%
  as.data.frame() %>%
  row_to_names(row_number = 1)
```
```{r}
faecal_metadata_row <- faecal_metadata %>%
  rownames_to_column(., "sample")

phylum_summary <- faecal_table %>%
  rownames_to_column(., "asv") %>%
  mutate_at(vars(-asv),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-asv, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(faecal_metadata_row, by = join_by(sample == sample)) %>% #append sample metadata
  left_join(., taxonomy_table, by = join_by(asv == asv)) %>% #append genome metadata
  group_by(sample,Phylum) %>%
  summarise(relabun=sum(count))

Phylum_arrange <- phylum_summary %>%
    group_by(Phylum) %>%
    summarise(mean=sum(relabun)) %>%
    arrange(-mean) %>%
    select(Phylum) %>%
    pull()

pdf("figures/phylum_faecal.pdf",width=14, height=9)
phylum_summary %>%
    left_join(taxonomy_table %>% 
                select(Phylum) %>% 
                unique(),
              by=join_by(Phylum==Phylum)) %>%
    left_join(faecal_metadata_row,by=join_by(sample==sample)) %>%
    filter(Phylum %in% Phylum_arrange[1:8]) %>%
    mutate(Phylum=factor(Phylum,levels=rev(Phylum_arrange[1:8]))) %>%
    filter(relabun > 0) %>%
    ggplot(aes(x=relabun, y=Phylum, group=Phylum, color=Phylum)) +
#        scale_color_manual(values=phylum_colors[-8]) +
        geom_jitter(alpha=0.5, show.legend = ) + 
        facet_grid(.~Working_status)+ 
         theme(panel.background = element_blank(),
               panel.grid.minor = element_line(size = 0.15, linetype = 'solid',
                                colour = "grey"),
          panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                colour = "grey"),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black"))+
        labs(y="Phylum", x="Relative abundance")
dev.off()
```
# Order percentages
```{r order_faecal, comment="", echo=FALSE}
physeq_faecal.order <- microbiome::aggregate_taxa(physeq_faecal, 'Order')
physeq_faecal.order.rel <-  microbiome::transform(physeq_faecal.order, "compositional")
faecal.order.rel <- physeq_faecal.order.rel@otu_table*100
means.faecal.order.rel <- as.data.frame(rowMeans(faecal.order.rel))
sd.faecal.order.rel <- as.data.frame(rowSds(faecal.order.rel, useNames = TRUE))
summary.order.faecal <- merge(means.faecal.order.rel, sd.faecal.order.rel, by="row.names")
colnames(summary.order.faecal) <- c("order","mean", "sd")
percentage.faecal.order.sort <- summary.order.faecal[order(-summary.order.faecal$mean),]
print(percentage.faecal.order.sort[1-10,], row.names = FALSE)
```

# Family percentages
```{r family_faecal, comment="", echo=FALSE}
physeq_faecal.family <- microbiome::aggregate_taxa(physeq_faecal, 'Family')
physeq_faecal.family.rel <-  microbiome::transform(physeq_faecal.family, "compositional")
faecal.family.rel <- physeq_faecal.family.rel@otu_table*100
means.faecal.family.rel <- as.data.frame(rowMeans(faecal.family.rel))
sd.faecal.family.rel <- as.data.frame(rowSds(faecal.family.rel, useNames = TRUE))
summary.family.faecal <- merge(means.faecal.family.rel, sd.faecal.family.rel, by="row.names")
colnames(summary.family.faecal) <- c("family","mean", "sd")
percentage.faecal.family.sort <- summary.family.faecal[order(-summary.family.faecal$mean),]
print(percentage.faecal.family.sort[1-10,], row.names = FALSE)
```

```{r}
faecal_family_rel <- physeq_faecal.family.rel@otu_table
family_mean <- faecal_family_rel %>%
  t() %>%
  as.data.frame()%>%
  rownames_to_column(., "sample") %>%
  left_join(., faecal_metadata_row, by = join_by(sample == sample)) %>%
  group_by(Working_status) %>%
  summarise_at(.vars = names(.)[2:63],.funs = c(mean="mean")) %>%
  t() %>%
  as.data.frame() %>%
  row_to_names(row_number = 1)
```

```{r}
faecal_metadata_row <- faecal_metadata %>%
  rownames_to_column(., "sample")

family_summary <- faecal_table %>%
  rownames_to_column(., "asv") %>%
  mutate_at(vars(-asv),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-asv, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(faecal_metadata_row, by = join_by(sample == sample)) %>% #append sample metadata
  left_join(., taxonomy_table, by = join_by(asv == asv)) %>% #append genome metadata
  group_by(sample,Family) %>%
  summarise(relabun=sum(count))

family_arrange <- family_summary %>%
    group_by(Family) %>%
    summarise(mean=sum(relabun)) %>%
    arrange(-mean) %>%
    select(Family) %>%
    pull()

pdf("figures/family_faecal.pdf",width=14, height=9)
family_summary %>%
    left_join(taxonomy_table %>% 
                select(Family,Phylum) %>% 
                unique(),
              by=join_by(Family==Family)) %>%
    left_join(faecal_metadata_row,by=join_by(sample==sample)) %>%
    filter(Family %in% family_arrange[1:20]) %>%
    mutate(Family=factor(Family,levels=rev(family_arrange[1:20]))) %>%
    filter(relabun > 0) %>%
    ggplot(aes(x=relabun, y=Family, group=Family, color=Phylum)) +
#        scale_color_manual(values=phylum_colors[-8]) +
        geom_jitter(alpha=0.5) + 
        facet_grid(.~Working_status)+
        theme_minimal() + 
        labs(y="Family", x="Relative abundance", color="Phylum")
dev.off()
```

```{r}
faecal_metadata_row <- faecal_metadata %>%
  rownames_to_column(., "sample")

Genus_summary <- faecal_table %>%
  rownames_to_column(., "asv") %>%
  mutate_at(vars(-asv),~./sum(.)) %>% #apply TSS normalisation
  pivot_longer(-asv, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(faecal_metadata_row, by = join_by(sample == sample)) %>% #append sample metadata
  left_join(., taxonomy_table, by = join_by(asv == asv)) %>% #append genome metadata
  group_by(sample,Genus) %>%
  summarise(relabun=sum(count))

Genus_arrange <- Genus_summary %>%
    group_by(Genus) %>%
    summarise(mean=sum(relabun)) %>%
    arrange(-mean) %>%
    select(Genus) %>%
    pull()

pdf("figures/Genus_faecal.pdf",width=14, height=9)
Genus_summary %>%
    left_join(taxonomy_table %>% 
                select(Genus,Phylum) %>% 
                unique(),
              by=join_by(Genus==Genus)) %>%
    left_join(faecal_metadata_row,by=join_by(sample==sample)) %>%
    filter(Genus %in% Genus_arrange[1:20]) %>%
    mutate(Genus=factor(Genus,levels=rev(Genus_arrange[1:20]))) %>%
    filter(relabun > 0) %>%
    ggplot(aes(x=relabun, y=Genus, group=Genus, color=Phylum)) +
#        scale_color_manual(values=phylum_colors[-8]) +
        geom_jitter(alpha=0.5) + 
        facet_grid(.~Working_status)+
        theme_minimal() + 
        labs(y="Genus", x="Relative abundance", color="Phylum")
dev.off()


```


# Genus percentages
```{r genus_faecal, comment="", echo=FALSE}
physeq_faecal.genus <- microbiome::aggregate_taxa(physeq_faecal, 'Genus')
physeq_faecal.genus.rel <-  microbiome::transform(physeq_faecal.genus, "compositional")
faecal.genus.rel <- physeq_faecal.genus.rel@otu_table*100
means.faecal.genus.rel <- as.data.frame(rowMeans(faecal.genus.rel))
sd.faecal.genus.rel <- as.data.frame(rowSds(faecal.genus.rel, useNames = TRUE))
summary.genus.faecal <- merge(means.faecal.genus.rel, sd.faecal.genus.rel, by="row.names")
colnames(summary.genus.faecal) <- c("family","mean", "sd")
percentage.faecal.genus.sort <- summary.genus.faecal[order(-summary.genus.faecal$mean),]
print(percentage.faecal.genus.sort[1-10,], row.names = FALSE)
```

# Beta diversity
```{r beta_div, comment="", echo=FALSE, message=FALSE, warning=FALSE, results='hide', eval=FALSE}
beta_q0 <- faecal_table %>%
  column_to_rownames(., "asv")%>%
  hillpair(., q = 0)

beta_q1n <- faecal_table %>%
  column_to_rownames(., "asv")%>%
  hillpair(., q = 1)

beta_q1p <- faecal_table%>%
  column_to_rownames(., "asv")%>%
  hillpair(., q = 1, tree = tree_faecal)

save(beta_q0,beta_q1n,beta_q1p, file = "resources/beta_div_faecal_filter.Rdata")#low abundance ASV removed
save(beta_q0,beta_q1n,beta_q1p, file = "resources/beta_div_faecal.Rdata")

```
```{r beta_div_load, comment="", echo=FALSE, message=FALSE, warning=FALSE}
load("resources/beta_div_faecal.Rdata")
load("resources/beta_div_faecal_filter.Rdata")
```

## Neutral beta diversity
```{r betan, comment="", echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
beta_metric <- beta_q1n$S

beta_q1n_nmds <- beta_metric %>%
  vegan::metaMDS(., trymax = 500, k = 2, verbosity = FALSE) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(faecal_metadata, by = join_by(sample == sample))

group_n <- length(unique(beta_q1n_nmds$Working_status))
beta_colors <- c("#e5bd5b", "#6b7398", "#76b183", "#d57d2c", "#2a2d26", "#f9d4cc", "#3c634e", "#ea68c3")
```
```{r beta_div_neutral, comment="", echo=FALSE, message=FALSE, warning=FALSE}
q1n_nmds <- beta_q1n_nmds %>%
  group_by(Working_status) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup()

#pdf("figures/neutral_beta_faecal.pdf",width=14, height=9)
ggplot(q1n_nmds, aes(x = NMDS1, y = NMDS2, color = Working_status)) +
  scale_color_manual(values = beta_colors[c(1:group_n)]) +
  scale_shape_manual(values = 1:10) +
  geom_point(size = 2) +
  #   stat_ellipse(aes(color = beta_q1n_nmds$Groups))+
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
  theme_classic() +
  theme(
    axis.text.x = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.title = element_text(size = 14, face = "bold"),
    axis.text = element_text(face = "bold", size = 12),
    panel.background = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 12),
    legend.title = element_text(size = 14),
    legend.position = "right", legend.box = "vertical"
  ) + labs(color='Role on the group')
#dev.off()
#+ geom_text(aes(label = Sample), size=8)
```

***Homogeneity of variance***

```{r permu1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ps.disper.neutral <- betadisper(beta_metric, faecal_metadata$Working_status) 
permutest(ps.disper.neutral, pairwise = TRUE) 
```

***Permanova***
```{r adonisbeta2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#### Adonis
metarow <- column_to_rownames(faecal_metadata, "sample")

adonis2(beta_metric ~ Group+Working_status+Sex, data = metarow[labels(beta_metric), ], permutations = 999) %>%
  as.matrix() %>%
  knitr::kable(., digits = c(0,3,3,3,3))

adonis2(beta_metric ~ Working_status, data = metarow[labels(beta_metric), ], permutations = 999, strata =metarow$Group) %>%
  as.matrix() %>%
  knitr::kable(., digits = c(0,4,4,4,4,4))

adonis2(beta_metric ~ Group+Working_status+Sex, data = metarow[labels(beta_metric), ], permutations = 999, strata =metarow$Group) %>%
  as.matrix() %>%
  knitr::kable(., digits = c(0,4,4,4,4,4))
```

## MAGs in different between followers and leaders

```{r upset, comment="", echo=FALSE, message=FALSE, warning=FALSE}
## Upset visualization: MAGs in different locations and shared among locations
library(UpSetR)
locationcolors=c('#c4d7d1','#408892')

faecal_table_g <- data.frame(physeq_faecal.genus@otu_table)
faecal_metadata_g <- data.frame(physeq_faecal.genus@sam_data)


Daneborg_samples <- faecal_metadata_g %>% 
  rownames_to_column(., "sample") %>% 
                    filter(Working_status == "Lead") %>% 
                    dplyr::select(sample) %>% pull()

Ittoq_samples <- faecal_metadata_g %>% 
  filter(Working_status == "Follower") %>% 
  rownames_to_column(., "sample") %>% 
  dplyr::select(sample) %>% pull()

genome_counts_row<-faecal_table_g
Ittoq <- genome_counts_row[, colnames(genome_counts_row) %in% Ittoq_samples]%>%
  rowSums()%>% as.data.frame()%>% dplyr::rename(Ittoq=".")%>%rownames_to_column(., "genome")
  
table_upset_analysis <- genome_counts_row[, colnames(genome_counts_row) %in% Daneborg_samples]%>%
  rowSums()%>% as.data.frame()%>% dplyr::rename(Daneb=".")%>%rownames_to_column(., "genome")%>%
      dplyr::left_join(., Ittoq, by = join_by(genome == genome))%>%column_to_rownames(., "genome")

table_upset_analysis_binary <- ifelse(table_upset_analysis > 0, 1, 0) %>% as.data.frame()

#pdf("figures/MAG_intersection.pdf",width=8,height=6, onefile=F)
upset(as.data.frame(table_upset_analysis_binary),
  keep.order = T,
#  sets = rev(c("feces","cloaca")),
  sets.bar.color= rev(locationcolors),
  mb.ratio = c(0.55, 0.45), order.by = "freq")
#dev.off()

```

## Ancombc2 (considering structural zeros)
```{r ancom_rand, comment="", echo=FALSE, message=FALSE, warning=FALSE}
library(ANCOMBC)
#x <- faecal_table %>% rownames_to_column(., "sample")%>% t()%>%#append genome metadata
#  left_join(., faecal_metadata, by = join_by(sample == sample))
set.seed(1234) #set seed for reproducibility
#fm <- lmer(Pool ~ Working_status + (1 | Group), data=a)
#taula <- data.frame(physeq_faecal@otu_table)
#metadata <- data.frame(physeq_FL@sam_data)
#physeq_faecal1<-physeq_faecal

#a$Team <- paste(a$Pack, "Team", sep = "")
#sample_tab <- sample_data(a)
#sample_data(physeq_faecal1)<- sample_tab

physeq_faecal_genus <- microbiome::aggregate_taxa(physeq_faecal, 'Genus')
ancom_rand_output_generaagglo = ancombc2(data = physeq_faecal_genus, 
                  assay_name = "counts",
                  tax_level = NULL, #change to agglomerate analysis to a higher taxonomic range
                  fix_formula = "Working_status", #fixed variable(s)
                  rand_formula = "(1|Group)",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0, 
                  s0_perc = 0,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lmerControl(),#lme4::lmerControl()
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)


ancom_rand_output_genera = ancombc2(data = physeq_faecal, 
                  assay_name = "counts",
                  tax_level = "Genus", 
                  fix_formula = "Working_status", 
                  rand_formula = "(1|Group)",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut = 0, 
                  s0_perc = 0,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

tax <- data.frame(physeq_faecal@tax_table) %>%
  rownames_to_column(., "taxon")

genus_unique <- tax %>%
  distinct(Genus,.keep_all = TRUE)

taxonomy_unique <- tax %>%
  distinct(Family,.keep_all = TRUE)

ancom_rand_output_genera_filt <- ancom_rand_output_genera$res %>%
  dplyr::select(taxon, lfc_Working_statusLead, p_Working_statusLead) %>%
  filter(p_Working_statusLead < 0.05) %>%
  dplyr::arrange(lfc_Working_statusLead) %>% 
  separate_wider_delim(taxon, ":", names = c('id', 'taxon')) %>%
  left_join(genus_unique, by=join_by(taxon==Genus))

ancom_rand_output_generaagglo_filt <- ancom_rand_output_generaagglo$res %>%
  dplyr::select(taxon, lfc_Working_statusLead, p_Working_statusLead) %>%
  filter(p_Working_statusLead < 0.05) %>%
  dplyr::arrange(lfc_Working_statusLead) %>% 
  left_join(genus_unique, by=join_by(taxon==Genus))


colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(tax, by=join_by(phylum == Phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table$Phylum))
colnames(tax_table)[1] <- "Phylum"

tax_color <- left_join(tax_table, colors_alphabetic, by=join_by(Phylum==phylum))%>%
	dplyr::arrange(Phylum) %>%
	dplyr::select(colors) %>%
	pull()
```


```{r ancombc_rand_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
ggplot(ancombc_rand_table2, aes(x=forcats::fct_rev(taxon), y=lfc_Working_statusLead, color=taxon)) + 
  geom_point(size=4) + 
#  scale_color_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10, face="bold.italic"),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
          panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                colour = "grey"))+
  xlab("Genus") + 
  ylab("log2FoldChange")+
  guides(col=guide_legend("Phylum"))

colors <- c("#e5bd5b", "#6b7398", "#76b183", "#d57d2c")#, "#2a2d26", "#f9d4cc", "#3c634e", "#ea68c3")
pdf("figures/differential_faecal_genus2.pdf",width=14, height=9)
ancom_rand_output_genera_filt %>%
  mutate(taxon=factor(taxon,levels=ancom_rand_output_genera_filt$taxon)) %>%
  ggplot(aes(x=lfc_Working_statusLead, y=forcats::fct_rev(taxon), fill=Phylum)) +
  geom_col(size = 2) +
  scale_fill_manual(values = colors) +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0, linetype="dashed", color = "grey", size=1) +
  geom_text(aes(1.5, 15), label = "Enriched\nin Leader", color="#666666") +
  geom_text(aes(-2.5, 5), label = "Enriched\nin Follower", color="#666666")+
  theme(axis.text.x = element_text(vjust = 0.5, size = 14),
        axis.text.y = element_text(size = 14),
        panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.title = element_text(size = 16, face = "bold"),
        axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        strip.background = element_blank(),
        strip.text = element_blank(),    
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14),
        ) +
        labs(x="log2FoldChange", y="Genera")
dev.off()
```

```{r}
#pdf("figures/differential_faecal_genus1.pdf",width=14, height=9)
ancom_rand_output_generaagglo_filt %>%
  mutate(taxon=factor(taxon,levels=ancom_rand_output_generaagglo_filt$taxon)) %>%
  ggplot(aes(x=lfc_Working_statusLead, y=forcats::fct_rev(taxon), fill=Phylum)) +
  geom_col(size = 2) +
          scale_fill_manual(values = colors) +
        geom_hline(yintercept = 0) +
        geom_vline(xintercept = 0, linetype="dashed", color = "grey", size=1) +
      geom_text(aes(1.5, 8), label = "Enriched\nin Leader", color="#666666") +
      geom_text(aes(-2.5, 5), label = "Enriched\nin Follower", color="#666666")+
        theme(
          panel.background = element_blank(),
          axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
          axis.title.x = element_blank(),
          strip.background = element_blank(), 
          strip.text = element_blank()
        ) +
        labs(x="log2FoldChange", y="Family")
#dev.off()
```

```{r ancom_rand_volcano, comment="", echo=FALSE, message=FALSE, warning=FALSE}
ancom_result <- ancom_rand_output1$res %>%
 # na.omit() %>%
#  dplyr::rename(genome=taxon) %>%
  left_join(tax,by=join_by(taxon==taxon))
ancom_result %>%
    mutate(significance = ifelse(p_Working_statusLead < 0.05, "1", "0")) %>%
    ggplot(., aes(x=-log(p_Working_statusLead), y=lfc_Working_statusLead, color=significance)) +
      geom_point() +
      scale_color_manual(values = c("#cccccc","#00FFFF")) +
      geom_text(aes(3, 5), label = "Enriched\nin Ittoqqortoormiit", color="#666666") +
      geom_text(aes(3, -5), label = "Enriched\nin Danenborg", color="#666666") +
      labs(color="Significance", y="Difference between locations", x="p-value") +
      theme_classic()
```

```{r}
physeq_L <- subset_samples(physeq_faecal, Working_status %in% c("Lead"))
physeq_L <- prune_taxa(taxa_sums(physeq_L)>0, physeq_L)
physeq_L.core <- core(physeq_L, detection = 0, prevalence = 0.90)
tax_table(physeq_L.core)
```

```{r}
physeq_FL.core <- core(physeq_faecal, detection = 0, prevalence = 0.99)
core_FL <- as.data.frame(tax_table(physeq_FL.core))
physeq_faecal.genus
physeq_FL.core_gen <- core(physeq_faecal.genus, detection = 0, prevalence = 0.90)
core_FL <- as.data.frame(tax_table(physeq_FL.core_gen))
```
```{r}
physeq_L <- subset_samples(physeq_faecal, Working_status %in% c("Lead"))
physeq_L <- prune_taxa(taxa_sums(physeq_L)>0, physeq_L)
physeq_L.core <- core(physeq_L, detection = 0, prevalence = 0.99)
tax_table(physeq_L.core)

physeq_L_gen <- subset_samples(physeq_faecal.genus, Working_status %in% c("Lead"))
physeq_L_gen <- prune_taxa(taxa_sums(physeq_L_gen)>0, physeq_L_gen)
physeq_L_gen.core <- core(physeq_L_gen, detection = 0, prevalence = 0.99)
tax_table(physeq_L_gen.core)
```

```{r}
physeq_F <- subset_samples(physeq_faecal, Working_status %in% c("Follower"))
physeq_F <- prune_taxa(taxa_sums(physeq_F)>0, physeq_F)
physeq_F.core <- core(physeq_F, detection = 0, prevalence = 0.99)
core_F <- as.data.frame(tax_table(physeq_F.core))

physeq_F_gen <- subset_samples(physeq_faecal.genus, Working_status %in% c("Follower"))
physeq_F_gen <- prune_taxa(taxa_sums(physeq_F_gen)>0, physeq_F_gen)
physeq_F_gen.core <- core(physeq_F_gen, detection = 0, prevalence = 0.90)
physeq_F_gen_F <- as.data.frame(tax_table(physeq_F_gen.core))
```


```{r}
physeq_faecal.genus <- microbiome::aggregate_taxa(physeq_faecal, 'Genus')

physeqG_FL.core <- core(physeq_faecal.genus, detection = 0, prevalence = 0.99)
coreG_FL <- as.data.frame(tax_table(physeqG_FL.core))

```
```{r}
physeqG_L <- subset_samples(physeq_faecal.genus, Working_status %in% c("Lead"))
physeqG_L <- prune_taxa(taxa_sums(physeqG_L)>0, physeqG_L)
physeqG_L.core <- core(physeqG_L, detection = 0, prevalence = 0.99)
coreG_L <- as.data.frame(tax_table(physeqG_L.core))

```

```{r}
physeqG_F <- subset_samples(physeq_faecal.genus, Working_status %in% c("Follower"))
physeqG_F <- prune_taxa(taxa_sums(physeqG_F)>0, physeqG_F)
physeqG_F.core <- core(physeqG_F, detection = 0, prevalence = 0.99)
coreG_F <- as.data.frame(tax_table(physeqG_F.core))
```
```{r}
library(RColorBrewer)
library(reshape)

prevalences <- seq(.05, 1, .05)

detections <- round(10^seq(log10(0.01), log10(.2), length = 9), 3)

# Also define gray color palette
gray <- gray(seq(0,1,length=5))

#Added pseq.rel, I thin... must be checked if it was in the the rednred version,; where it is initialized
pseq.rel<- microbiome::transform(physeq_faecal.genus, 'compositional')
#min-prevalence gets the 100th highest prevalence
p <- plot_core(pseq.rel,
               plot.type = "heatmap", 
               colours = rev(brewer.pal(5, "Spectral")),
               prevalences = prevalences, 
               detections = detections, 
               min.prevalence = prevalence(pseq.rel, sort = TRUE)[100]) +
  labs(x = "Detection Threshold\n(Relative Abundance (%))") +
    
  #Adjusts axis text size and legend bar height
  theme(axis.text.y= element_text(size=8, face="italic"),
        axis.text.x.bottom=element_text(size=8),
        axis.title = element_text(size=10),
        legend.text = element_text(size=8),
        legend.title = element_text(size=10))

print(p)
```
```{r}
physeq_F <- subset_samples(physeq_faecal.genus, Working_status %in% c("Follower"))
physeq_F <- prune_taxa(taxa_sums(physeq_F)>0, physeq_F)
pseq.rel<- microbiome::transform(physeq_F, 'compositional')
#min-prevalence gets the 100th highest prevalence
p <- plot_core(pseq.rel,
               plot.type = "heatmap", 
               colours = rev(brewer.pal(5, "Spectral")),
               prevalences = prevalences, 
               detections = detections, 
               min.prevalence = prevalence(pseq.rel, sort = TRUE)[100]) +
  labs(x = "Detection Threshold\n(Relative Abundance (%))") +
    
  #Adjusts axis text size and legend bar height
  theme(axis.text.y= element_text(size=8, face="italic"),
        axis.text.x.bottom=element_text(size=8),
        axis.title = element_text(size=10),
        legend.text = element_text(size=8),
        legend.title = element_text(size=10))

print(p)
```
```{r}
physeq_L <- subset_samples(physeq_faecal.genus, Working_status %in% c("Lead"))
physeq_L <- prune_taxa(taxa_sums(physeq_L)>0, physeq_L)
pseq.rel<- microbiome::transform(physeq_L, 'compositional')
#min-prevalence gets the 100th highest prevalence
p <- plot_core(pseq.rel,
               plot.type = "heatmap", 
               colours = rev(brewer.pal(5, "Spectral")),
               prevalences = prevalences, 
               detections = detections, 
               min.prevalence = prevalence(pseq.rel, sort = TRUE)[69]) +
  labs(x = "Detection Threshold\n(Relative Abundance (%))") +
    
  #Adjusts axis text size and legend bar height
  theme(axis.text.y= element_text(size=8, face="italic"),
        axis.text.x.bottom=element_text(size=8),
        axis.title = element_text(size=10),
        legend.text = element_text(size=8),
        legend.title = element_text(size=10))

print(p)
```
```{r}
physeq_faecal.genus.rel <-  microbiome::transform(physeq_faecal.genus, "compositional")
ex3 <- subset_taxa(physeq_faecal.genus.rel, Genus %in% c("Bacteroides","Clostridium_sensu_stricto_1", "Blautia", "Lachnoclostridium"))
gen_sam<- data.frame(ex3@sam_data)%>%
  rownames_to_column(., "sample")
gen<- data.frame(ex3@otu_table)%>%
  t() %>%
  as.data.frame()%>%
  rownames_to_column(., "sample") %>%
  left_join(., gen_sam, by = join_by(sample == sample)) %>%
  group_by(Working_status) %>%
  summarise_at(.vars = names(.)[2:63],.funs = c(mean="mean")) %>%
  t() %>%
  as.data.frame() %>%
  row_to_names(row_number = 1)


colNames <- names(gen)[2:5]
for(i in colNames){
  plt <- ggplot(gen, aes(x=Working_status, y=.data[[i]], color = Working_status)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank())
print(plt)
}
```


```{r}
alpha_colors <- c("#e5bd5b", "#6b7398", "#76b183", "#d57d2c", "#2a2d26", "#f9d4cc", "#3c634e", "#ea68c3")
group_n <- gen %>%
  select(Group) %>%
  pull() %>%
  unique() %>%
  length()

highlight_df <- gen %>% 
             filter(Working_status=="Lead")
gen %>%
  ggplot(aes(x = Group, y = Lachnoclostridium, group = Group, color = Group, fill = Group)) +
   geom_boxplot(alpha = 0.2, outlier.shape = NA, show.legend = FALSE)+
   geom_jitter(width = 0.2, size = 2.5, show.legend = FALSE) + 
  geom_point(data=highlight_df, 
             aes(x=Group,y=Lachnoclostridium), 
             color='red',
             size=4, show.legend = FALSE)+
#  geom_violin(alpha = 0.2, width = 0.3, show.legend = FALSE) +
  scale_color_manual(values = alpha_colors[c(1:group_n)]) +
  scale_fill_manual(values = paste0(alpha_colors[c(1:group_n)], "50")) +
#  stat_compare_means(show.legend = FALSE) +
  theme(axis.text.x = element_text(vjust = 0.5, size = 14),
        axis.text.y = element_text(size = 14),
        axis.title = element_text(size = 16, face = "bold"),
        axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
        axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)),
        panel.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.title = element_text(size=22, face="bold"),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black")) +
  guides(colour = guide_legend(override.aes = list(size = 6))) +
  labs(x = "Sled dog group", y = "Neutral alpha diversity", title="Bacteroides")
```

















